### DISCUSSION TITLE
Rich, you said on HN, 'both runes and non-runes mode will be supported for the next two major versions', so will it be a breaking change at some point?

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#10768] [Q&A] [answered]
Or will we be able to use both forever? The video made this impression. Not arguing about what's better, rune or non-rune, just seeing thousands LOC I need to refactor then and not sure how to write new components, _now_.

**Comment by geoffrich** [comment, marked as answer]
Current plan is to _eventually_ deprecate the current reactivity system in favor of Runes.

> [The old way] will be supported for 5 and 6, but _probably_ removed in 7. In other words you have all the time in the world (and there'll be automatic migration tooling)

https://twitter.com/Rich_Harris/status/1705218636024008830

From the [FAQ](https://svelte-5-preview.vercel.app/faq):

> Eventually, you'll have to make some changes — most of which we hope to automate. We don't want to end up in a situation where people feel like they have to juggle knowledge of a bunch of different ways of doing things.
> 
> Our current plan is that some or all of the features that runes make unnecessary will be deprecated in Svelte 6 and removed in Svelte 7. But don't worry — that won't happen for some time, and we'll provide automatic migration tooling to do as much of the change as possible.

**Reply by 205g0** [reply]
I wonder if also get, writable, derived are affected...

**Reply by teemingc** [reply]
stores will still be supported (although in some cases it would benefit you to refactor using runes)

**Reply by 205g0** [reply]
> stores will still be supported

Awesome, otherwise I wouldn't know how to refactor. I have many, many custom stores which are interweaved and whatever. I think that custom stores actually are the true power of Svelte.

> would benefit you to refactor using runes)

Yeah, one reason would be that you have  one style for both .svelte and .js/.ts files and in general, just one way of doing thinhgs.

However, I'm not so sure if the ergonomics of runes beat the one of custom stores. But let's see. Most important is that we can keep traditional stores and they are not gonna deprecated.
